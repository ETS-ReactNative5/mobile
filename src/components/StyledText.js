import React from 'react'
import {
  Text
} from 'react-native'
import EStyleSheet from 'react-native-extended-stylesheet'
import { append } from 'ramda'

const StyledText = (props) => {
  let textStyle = ( props.textStyle ? [styles.defaultText, styles[props.textStyle]] : [styles.defaultText] )
  textStyle = (props.additionalStyles ? append(props.additionalStyles, textStyle) : textStyle)

  const { numberOfLines, ellipsizeMode } = props
  const ellipsisProps = (props.numberOfLines > 0 ? {numberOfLines, ellipsizeMode} : null)

  return (
    <Text style={textStyle} {...ellipsisProps}>
      {props.text}
    </Text>
  )
}

const styles = EStyleSheet.create({
  defaultText: {
    fontFamily: 'OpenSans',
    fontSize: 14,
  },
  headerText: {
    fontSize: 18,
    letterSpacing: 1.5,
    marginBottom: 5
  },
  errorMessage: {
    color: 'red',
    marginBottom: 10,
    lineHeight: 24
  },
  link: {
    color: '$headerColor'
  },
  smallLink: {
    fontSize: 12,
    color: '$headerColor'
  },
  small: {
    fontSize: 12,
  },
  largeLink: {
    color: '$darkTextColor',
    fontSize: 20,
  },
  large: {
    fontSize: 20,
  },
  largeBold: {
    fontSize: 20,
    fontWeight: 'bold'
  },
  citation: {
    color: 'grey',
    fontStyle: 'italic',
    fontSize: 12,
    flex: 1,
  },
  fullHeight: {
    height: '100%'
  },
  subLabelText: {
    fontSize: 12,
    fontStyle: 'italic',
    color: '$grey',
  }
});

StyledText.propTypes = {
  text: React.PropTypes.string,
  textStyle: React.PropTypes.string,
  additionalStyles: React.PropTypes.array,
  numberOfLines: React.PropTypes.number,
  ellipsizeMode: React.PropTypes.string,
}

export default StyledText
