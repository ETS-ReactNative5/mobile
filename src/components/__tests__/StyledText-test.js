import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import StyledText from '../StyledText'

it('renders text correctly', () => {
  const tree = renderer.create(
    <StyledText text={'O Hai!'} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders styles as array when style passed in', () => {
  const tree = renderer.create(
    <StyledText text={'O Hai!'}  textStyle={'test'} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
