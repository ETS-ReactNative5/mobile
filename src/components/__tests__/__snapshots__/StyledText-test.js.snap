exports[`test renders styles as array when style passed in 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      undefined,
      undefined,
    ]
  }>
  O Hai!
</Text>
`;

exports[`test renders text correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      undefined,
    ]
  }>
  O Hai!
</Text>
`;
